idl/github.com/uber/cadence/shared.thrift

```diff
@@ -835,3 +826,10 @@ struct WorkflowQuery {
   10: optional string queryType
   20: optional binary queryArgs
 }
+
+struct RespondQueryTaskCompletedRequest {
```

Comment:
It is ok to have single API for RespondQueryTaskCompleted but we should create different structs for communicating success and failures.

Reply:
will add an enum type for the QueryTaskCompletedType